lib/apt/spy2.rb

```diff
@@ -121,6 +121,42 @@ def filter(mirrors, strict = false, output = true)
 
   end
 
+  private
+  # architecture is not reported in the strings we need
+  def fix_arch(str)
+    if str == "x86_64"
+      return "amd64"
+    end
+
+    return "i386"
+  end
+
+  private
+  def broken?(url)
```

Comment:
So this is a slightly more sophisticated version of `URI.open()` which does a `HEAD` request and evaluates the status code. I also noticed that apparently some mirrors have broken SSL certs 🙈 and caught all exceptions accordingly.

Reply:
This is good!